app/src/main/java/de/danoeh/antennapod/dialog/ShareDialog.java

```diff
@@ -0,0 +1,121 @@
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.util.ShareUtils;
+
+public class ShareDialog extends DialogFragment {
+
+    private static final String TAG = "ShareDialog";
+    private final Context ctx;
+    private FeedItem item;
+
+    private RadioGroup radioGroup;
+    private RadioButton radioEpisodeWebsite;
+    private RadioButton radioMediaFile;
+    private Switch switchStartAt;
+    private SharedPreferences prefs;
+
+    public ShareDialog(Context ctx, FeedItem item) {
+        this.ctx = ctx;
+        this.item = item;
+        prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        View content = View.inflate(ctx, R.layout.share_episode_dialog, null);
+        AlertDialog.Builder builder = new AlertDialog.Builder(ctx);
+        builder.setTitle(R.string.share_episode_label);
+        builder.setView(content);
+
+        radioGroup = content.findViewById(R.id.share_dialog_radio_group);
+        radioEpisodeWebsite = content.findViewById(R.id.share_episode_website_radio);
+        radioMediaFile = content.findViewById(R.id.share_media_file_radio);
+        switchStartAt = content.findViewById(R.id.share_start_at_timer_dialog);
+
+        setupOptions();
+
+        builder
+                .setPositiveButton(R.string.share_episode_positive_label_button, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        if (radioEpisodeWebsite.isChecked()) {
+                            if (switchStartAt.isChecked()) {
+                                ShareUtils.shareFeedItemLink(ctx, item, true);
+                                prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_START_AT, true).apply();
+                            } else {
+                                ShareUtils.shareFeedItemLink(ctx, item);
+                                prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_START_AT, false).apply();
+                            }
+                            prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_WEBSITE, true).apply();
+                            prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_MEDIA, false).apply();
+                        } else {
+                            if (switchStartAt.isChecked()) {
+                                ShareUtils.shareFeedItemDownloadLink(ctx, item, true);
+                                prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_START_AT, true).apply();
+                            } else {
+                                ShareUtils.shareFeedItemDownloadLink(ctx, item);
+                                prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_START_AT, false).apply();
+                            }
```

Comment:
Can you please shorten this? These checks look pretty big and I think they can be simplified, eg. something like this:

```
ShareUtils.shareFeedItemDownloadLink(ctx, item, switchStartAt.isChecked());
prefs.edit().putBoolean(UserPreferences.PREF_SHARE_EPISODE_START_AT, switchStartAt.isChecked()).apply();
```

Reply:
sure, I have already updte this for the next commit. Thanks for this observation